### PR TITLE
feat(o-shop): login with manual user update

### PR DESCRIPTION
**Pull request by coffee-intocode**

### CODE REVIEW

File: src/store/login/reducers.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import actions from './actionTypes';
+
+let user = JSON.parse(localStorage.getItem('users'));
```
Comment (coffee-intocode) [inline src/store/login/reducers.js]: I think this should be updated to JSON.parse(localStorage.getItem('users')); in the sheet.

File: src/utils/fake-backend.js [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+    username: 'ahmed.aly',
+    password: 'umbrage',
+    id: 1,
+  },
+];
+
+let users_serialized = JSON.stringify(usersArray);
+
+localStorage.setItem('users', users_serialized);
+
```
Comment (coffee-intocode) [inline src/utils/fake-backend.js]: Im not sure if this is right but this is the only way I could get past the login screen.

File: src/utils/history.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+import { createBrowserHistory } from 'history';
```
Comment (coffee-intocode) [inline src/utils/history.js:1]: I kept seeing an error in the console so I looked into the docs and found this. Let me know what you think. 

https://github.com/ReactTraining/react-router/blob/master/FAQ.md#how-do-i-access-the-history-object-outside-of-components

File: src/utils/fake-backend.js [hunk 4]
```diff
@@ -1,4 +1,19 @@
 // array in local storage for registered users
+
+let usersArray = [
+  {
+    firstName: 'ahmed',
+    lastName: 'aly',
+    username: 'ahmed.aly',
+    password: 'umbrage',
```
Comment (ahaly92) [inline src/utils/fake-backend.js:8]: You can set this in local storage in your browser manually but this works too. We’ll end up ripping this out later anyway so this is good for now

File: src/utils/history.js (hunk 3 above)
Comment (ahaly92) [reply src/utils/history.js:1]: Yea you figured it out perfectly. I missed this file and will update for the future 🙏

**Review by ahaly92 — APPROVED** [review]
Nice work here!

File: src/store/login/reducers.js (hunk 1 above)
Comment (ahaly92) [reply src/store/login/reducers.js]: This should actually be user. When you register you add a user to your local storage entry labeled users and when you log in you are checking if the user is in users. If it is then you save their info in user. Happy to hop on a call to explain

File: src/utils/fake-backend.js (hunk 2 above)
Comment (ahaly92) [reply src/utils/fake-backend.js]: let's hop on a call to discuss

File: src/services/user-service.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+      return user;
+    });
+}
+function logout() {
+  localStorage.removeItem('user');
+}
+
+function register(user) {
+  //my code is going to go in here
```
Comment (BarretteBanner) [inline src/services/user-service.js]: let's delete this function for now since it's not currently being used

File: src/store/login/action.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  return (dispatch) => {
+    dispatch({ type: actionTypes.LOGIN_REQUEST, user: { username } });
+
+    userService.login(username, password).then(
+      (user) => {
+        dispatch({ type: actionTypes.LOGIN_SUCCESS, user });
+        history.push('/');
+      },
+      (error) => {
+        console.log(error);
```
Comment (BarretteBanner) [inline src/store/login/action.js]: no console.logs

File: src/utils/fake-backend.js [hunk 7]
```diff
@@ -1,4 +1,19 @@
 // array in local storage for registered users
```
Comment (BarretteBanner) [inline src/utils/fake-backend.js]: no comments

File: src/utils/fake-backend.js [hunk 8]
```diff
@@ -1,4 +1,19 @@
...
+
+let usersArray = [
```
Comment (BarretteBanner) [inline src/utils/fake-backend.js]: should use const

File: src/utils/fake-backend.js [hunk 9]
```diff
@@ -1,4 +1,19 @@
...
+  {
+    firstName: 'ahmed',
+    lastName: 'aly',
+    username: 'ahmed.aly',
+    password: 'umbrage',
+    id: 1,
+  },
+];
+
+let users_serialized = JSON.stringify(usersArray);
```
Comment (BarretteBanner) [inline src/utils/fake-backend.js]: same thing here with const

**Pull request closed without merging**